core/src/main/scala/scalaz/Nondeterminism.scala

```diff
@@ -62,6 +62,9 @@ trait Nondeterminism[F[_]] extends Monad[F] { self =>
 
   def chooseAny[A](head: F[A], tail: Seq[F[A]]): F[(A, Seq[F[A]])]
 
+  // TODO: Reimplement chooseAny using a more general collections method and then get rid of the toList call here.
+  def chooseAny[A](head: F[A], tail: IList[F[A]]): F[(A, Seq[F[A]])] = chooseAny(head, tail.toList)
```

Comment:
I fear a bit for overloading ambiguities. Maybe we should leave this one out until the `TODO` is resolve (could https://hackage.haskell.org/package/ListLike-4.5.1/docs/Data-ListLike.html be something to consider?)

Reply:
I (poorly) approached this a little bit in another branch mentioned in #1462.  @fommil pointed me to another issue (#1452) which, after some discussion, has some better ideas about the whole thing. 

I wanted to make the `IList` overload here to make `IList` more useful.  One of the reasons I've personally avoided `IList`, is that there are methods like this which work for the standard library but not IList, so  you wind up moving away from an `IList` anyway. 

After thinking about it a bit more, that's not even really a good argument for my overload above, since it winds up returning a `Seq`. I'll remove it for now, until #1452 has been looked in to.  I don't think `Foldable` + `Reducer` would help here.  I haven't looked in to it personally, but the rest of `Nondeterminism` is implemented in terms of those typeclasses.  I suspect the original author tried and found it not possible.